src/common/ownsql.cpp

```diff
@@ -190,7 +190,9 @@ QString SqlDatabase::error() const
 void SqlDatabase::close()
 {
     if (_db) {
-        for (auto *q : qAsConst(_queries)) {
+        // we need a copy because query is removed from a set in finish()
+        const auto queries = _queries;
```

Comment:
how did you spot this one?

Reply:
The tests were failed with the first commit. And after brief investigation I found this code
```
void SqlQuery::finish()
{
    ...
    if (_sqldb) {
        _sqldb->_queries.remove(this);
    }
}
```